### ISSUE TITLE
Flaky spec:  Legislation process page proposals phase create proposal button leads to register path if user is not logged in

### ISSUE THREAD

**Issue by bertocq** [AyuntamientoMadrid/consul#1233]
# What
Tests that fail randomly are called "flakies", this one seems to be one:

**Randomized seed:** 14499

**Travis failed build:** https://travis-ci.org/AyuntamientoMadrid/consul/jobs/338862099

**Failure:**

```
    2) Legislation process page proposals phase create proposal button leads to register path if user is not logged in
     Failure/Error: expect(page).to have_link("Create a proposal", href: new_user_session_path)
       expected to find visible link "Create a proposal" but there were no matches
     # ./spec/features/legislation/processes_spec.rb:237:in `block (4 levels) in <top (required)>'
```

# How
- [ ] Explain why the test is flaky, or under which conditions/scenario it fails randomly
- [ ] Explain why your PR fixes it
- [ ] Create a backport PR to consul/consul when the fixing PR is approved

### Random values issues
If the problem comes from randomly generated values, running multiple times a single spec could help you reproduce the failure by running at your command line:
```bash
for run in {1..10}
do
  bin/rspec ./spec/features/budgets/investments_spec.rb:256
done
```

You can also try running a single spec in Travis:
Add option `:focus` to the spec and push your branch to Github, for example:
```ruby
scenario 'Show', :focus do
```

But remember to remove that `:focus` changes afterwards when submitting your PR changes!

### Test order issues
Running specs in the order they failed may discover that the problem is that a previous test sets an state in the test environment that makes our flaky fail/pass. Tests should be independent from the rest.

After executing rspec you can see the seed used, add it as an option to rspec, for example:
`bin/rspec --seed 55638` (check **Randomized seed** value at beginning of issue)

### Other things to watch for
- Time related issues (current time, two time or date comparisons with miliseconds/time when its not needed)
- https://semaphoreci.com/community/tutorials/how-to-deal-with-and-eliminate-flaky-tests

**Comment by javierm** [comment]
Here's a recent build with the same failure: [Forked repo build 26, job 10](https://travis-ci.org/javierm/consul/jobs/421598691).

@decabeza I think I'll need your help with this one. The line causing the failure is the condition `if @process.id == 24` added in 60a5bb32. Do you have any clues about how we can find a more solid condition?

**Comment by javierm** [comment]
Closed via #1627.
